src/Features/Core/Portable/Completion/Providers/AbstractObjectCreationCompletionProvider.cs

```diff
@@ -116,22 +88,19 @@ protected override Task<ImmutableArray<ISymbol>> GetSymbolsAsync(SyntaxContext c
                     type.TypeKind == TypeKind.Dynamic ||
                     type.IsAbstract)
                 {
-                    return SpecializedTasks.EmptyImmutableArray<ISymbol>();
+                    return SpecializedTasks.EmptyImmutableArray<(ISymbol symbol, bool preselect)>();
                 }
 
-                if (type.TypeKind == TypeKind.TypeParameter &&
-                    !((ITypeParameterSymbol)type).HasConstructorConstraint)
-                {
-                    return SpecializedTasks.EmptyImmutableArray<ISymbol>();
-                }
+                if (type is ITypeParameterSymbol typeParameter && !typeParameter.HasConstructorConstraint)
+                    return SpecializedTasks.EmptyImmutableArray<(ISymbol symbol, bool preselect)>();
             }
 
             if (!type.IsEditorBrowsable(options.GetOption(RecommendationOptions.HideAdvancedMembers, context.SemanticModel.Language), context.SemanticModel.Compilation))
-            {
-                return SpecializedTasks.EmptyImmutableArray<ISymbol>();
-            }
+                return SpecializedTasks.EmptyImmutableArray<(ISymbol symbol, bool preselect)>();
 
-            return Task.FromResult(ImmutableArray.Create((ISymbol)type));
+            // In the case of array creation, we don't offer a preselected/hard-selected item because
+            // the user may want an implicitly-typed array creation
+            return Task.FromResult(ImmutableArray.Create(((ISymbol)type, preselect: !isArray)));
```

Comment:
this is an example of code we would run twice.  once for non-preselect, and once for preselect.  Now we just run things once, but determien if we should preselect or not just once.